ui/de-lib/src/main/java/org/iplantc/de/diskResource/client/presenters/metadata/MetadataPresenterImpl.java

```diff
@@ -217,6 +222,16 @@ public static Avu newMetadata(String attr, String value, String unit) {
         return avu;
     }
 
+    @Override
+    public Avu  setAvuModelKey(Avu avu) {
+        if (avu != null) {
+            final AutoBean<Avu> avuBean = AutoBeanUtils.getAutoBean(avu);
+            avuBean.setTag(AVU_BEAN_TAG_MODEL_KEY, String.valueOf(unique_avu_id++));
+            return avuBean.as();
```

Comment:
I'm on the fence about where to set the model keys.

It seems like it's actually logic that belongs to the view, since the presenter could potentially use a different view that doesn't care about list stores or model keys.

Just food for thought and I don't think this needs to change for this PR.